src/Workflow/MoveJatsAssets.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentStore\Workflow;
+
+use FluentDOM\DOM\Element;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Psr7\UriResolver;
+use InvalidArgumentException;
+use League\Flysystem\AdapterInterface;
+use League\Flysystem\FilesystemInterface;
+use Libero\ContentApiBundle\Model\PutTask;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\File\MimeType\ExtensionGuesser;
+use Symfony\Component\Workflow\Event\Event;
+use UnexpectedValueException;
+use function count;
+use function explode;
+use function GuzzleHttp\Promise\all;
+use function GuzzleHttp\Psr7\uri_for;
+use function implode;
+use function in_array;
+use function is_callable;
+use function is_resource;
+use function Libero\ContentApiBundle\stream_hash;
+use function sprintf;
+
+final class MoveJatsAssets implements EventSubscriberInterface
+{
+    private $client;
+    private $filesystem;
+
+    public function __construct(FilesystemInterface $filesystem, ClientInterface $client)
+    {
+        $this->filesystem = $filesystem;
+        $this->client = $client;
+
+        if (!is_callable([$filesystem, 'getHttpUrl'])) {
+            throw new InvalidArgumentException('Requires the HTTP URL plugin');
+        }
+    }
+
+    public static function getSubscribedEvents() : array
+    {
+        return [
+            'workflow.libero.content_store.put.transition.manipulate' => 'onManipulate',
+        ];
+    }
+
+    public function onManipulate(Event $event) : void
+    {
+        /** @var PutTask $task */
+        $task = $event->getSubject();
+        $document = $task->getDocument();
+
+        $document->registerNamespace('jats', 'http://jats.nlm.nih.gov');
+        $document->registerNamespace('xlink', 'http://www.w3.org/1999/xlink');
+
+        /** @var array<Element> $assets */
+        $assets = $document('(//jats:graphic|//jats:inline-graphic)[@xlink:href]');
```

Comment:
Wonder if you should have to set a whitelist of base URIs to rewrite, rather than downloading from anywhere?

Reply:
Means this can just look for _anything_ with a `@xlink:href` inside a `<jats:article>` and rewrite it as long as the absolute URI matches the whitelist.

Reply:
Hmm, more elements than I thought can have `@xlink:href`.